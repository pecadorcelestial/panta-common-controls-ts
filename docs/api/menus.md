# Menu with a blur effect

As its name suggests, this component consists in a ``header``, the ``options`` and a ``content`` layout where you place the components for the page, a wrapper kind of thing. When the menu is displayed, all the components inside the ``content`` section with be blured.

### Properties

| Name      | Type              | Required | Comments |
| ----------|-------------------|----------|----------|
| options   | array of elements | true     | An array of elements (nodes) that will be displayed in the menu. |
| title     | string | true     | Sets the _title_ for the ``header`` section. |

## Methods

| Name | Parameters | Type | Description |
|------|------------|------|-------------|
| show | N/A        | N/A  | Shows the menu... obviously. |
| hide | N/A        | N/A  | Hides the menu... also obvious. |

## Usage

```javascript
import React, { Component } from 'react';
import { MenuWithBlur } from 'panda-common-controls';

class BasicForm extends Component {
    render() {
        //Menu options.
        const options = [];
        options.push(<a href='/'>Home</a>);
        options.push(<a href='/second'>Second page</a>);
        options.push(<a href='/storybook-1'>Storybook vol. 1</a>);
        options.push(<a href='/storybook-2'>Storybook vol. 2</a>);
        options.push(<a href='/storybook-3'>Storybook vol. 3</a>);
        //Result.
        return(
            <MenuWithBlur title='Title' options={options}>
                <div>
                    {/* The rest of your content... */}
                </div>
            </MenuWithBlur>
        );
    }
}
```
