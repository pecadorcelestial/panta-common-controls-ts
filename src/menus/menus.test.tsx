
//RRRR  EEEEE  AAA   CCCC TTTTT
//R   R E     A   A C       T
//RRRR  EEE   AAAAA C       T
//R   R E     A   A C       T
//R   R EEEEE A   A  CCCC   T

import React from 'react';
import renderer from 'react-test-renderer';

//EEEEE N   N ZZZZZ Y   Y M   M EEEEE
//E     NN  N    Z   Y Y  MM MM E
//EEE   N N N   Z     Y   M M M EEE
//E     N  NN  Z      Y   M   M E
//EEEEE N   N ZZZZZ  YYY  M   M EEEEE

import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({ adapter: new Adapter() });

//JJJJJ EEEEE  SSSS TTTTT
//    J E     S       T
//    J EEE    SSS    T
//J   J E         S   T
// JJJ  EEEEE SSSS    T

//NOTA: Éste módulo ayuda a que los errores en componentes estilizados se vean de manera correcta:
//https://medium.com/styled-components/effective-testing-for-styled-components-67982eb7d42b
import 'jest-enzyme';
import 'jest-styled-components';

// CCCC  OOO  M   M PPPP   OOO  N   N EEEEE N   N TTTTT EEEEE  SSSS
//C     O   O MM MM P   P O   O NN  N E     NN  N   T   E     S
//C     O   O M M M PPPP  O   O N N N EEE   N N N   T   EEE    SSS
//C     O   O M   M P     O   O N  NN E     N  NN   T   E         S
// CCCC  OOO  M   M P      OOO  N   N EEEEE N   N   T   EEEEE SSSS

import { MenuWithBlur } from './menus';

//Snapshots (MENÚ CON BLUR)(Tema: default).
describe('[SNAPSHOT][Componentes][Common][MenuWithBlur] - Menú con efecto "blur" sobre el contenido.', () => {
    it('Debe pintar correctamente el menú con las opciones configuradas.', () => {
        let options: Array<any> = [];
        options.push(<label>Inicio</label>);
        options.push(<label>Información de usuario</label>);
        options.push(<label>Storybook vol. 1</label>);
        options.push(<label>Storybook vol. 2</label>);
        options.push(<label>Storybook vol. 3</label>);
        const child = <div style={{ backgroundColor: '#8394DE', height: '100px', width: '100px' }}/>;
        const component = renderer.create(<MenuWithBlur title='Title' options={options}>{child}</MenuWithBlur>).toJSON();
		expect(component).toMatchSnapshot();
    });
});

//Snapshots (MENÚ CON BLUR)(Tema: IENTC... (¬_¬')).
describe('[SNAPSHOT][Componentes][Common][MenuWithBlur] - Menú con efecto "blur" y tema "IENTC" sobre el contenido.', () => {
    it('Debe pintar correctamente el menú con las opciones configuradas.', () => {
        let options: Array<any> = [];
        options.push(<label>Inicio</label>);
        options.push(<label>Información de usuario</label>);
        options.push(<label>Storybook vol. 1</label>);
        options.push(<label>Storybook vol. 2</label>);
        options.push(<label>Storybook vol. 3</label>);
        const child = <div style={{ backgroundColor: '#8394DE', height: '100px', width: '100px' }}/>;
        const component = renderer.create(<MenuWithBlur title='Title' options={options} theme='IENTC'>{child}</MenuWithBlur>).toJSON();
		expect(component).toMatchSnapshot();
    });
});

//Métodos.
describe("[MÉTODOS][Componentes][Common][MenuWithBlur] - Valida las llamadas a los métodos dentro del componente.", () => {
	//Se crea el componente.
	//NOTA [1]: Al utilizar la función "mount" se detona la función "componentDidMount".
    //NOTA [2]: Además, con "mount" se puede utilizar el método "attachTo" para utilizar "document".
    let options: Array<any> = [];
    options.push(<label>Inicio</label>);
    options.push(<label>Información de usuario</label>);
    options.push(<label>Storybook vol. 1</label>);
    options.push(<label>Storybook vol. 2</label>);
    options.push(<label>Storybook vol. 3</label>);
    const child = <div style={{ backgroundColor: '#8394DE', height: '100px', width: '100px' }}/>;
    const component = enzyme.mount(<MenuWithBlur title='Title' options={options}>{child}</MenuWithBlur>);
	
	// CCCC  OOO  M   M PPPP   OOO  N   N EEEEE N   N TTTTT DDDD  IIIII DDDD  M   M  OOO  U   U N   N TTTTT
	//C     O   O MM MM P   P O   O NN  N E     NN  N   T   D   D   I   D   D MM MM O   O U   U NN  N   T
	//C     O   O M M M PPPP  O   O N N N EEE   N N N   T   D   D   I   D   D M M M O   O U   U N N N   T
	//C     O   O M   M P     O   O N  NN E     N  NN   T   D   D   I   D   D M   M O   O U   U N  NN   T
	// CCCC  OOO  M   M P      OOO  N   N EEEEE N   N   T   DDDD  IIIII DDDD  M   M  OOO   UUU  N   N   T

	it('Se verifica que la información inicial sea correcta.', () => {
		//Expectativas.
        expect(component.state().show).toBe(false);
        expect(component.state().hide).toBe(false);
	});

    // SSSS H   H  OOO  W   W
    //S     H   H O   O W   W
    // SSS  HHHHH O   O W W W
    //    S H   H O   O WW WW
    //SSSS  H   H  OOO  W   W

    it('Debe detonar la animación de entrada.', () => {
        //Simulación.
        component.instance().show();
        //Expectativas.
		expect(component.state().show).toBe(true);
        expect(component.state().hide).toBe(false);
    });

    //H   H IIIII DDDD  EEEEE
    //H   H   I   D   D E
    //HHHHH   I   D   D EEE
    //H   H   I   D   D E
    //H   H IIIII DDDD  EEEEE

    it('Debe detonar la animación de salida.', (done) => {
        //Simulación.
        component.instance().hide();
        //Expectativas.
        setTimeout(() => {
            done();
            //Expectativa.
            expect(component.state().show).toBe(false);
            expect(component.state().hide).toBe(true);
        }, 1500);
    });
    
	//U   U N   N M   M  OOO  U   U N   N TTTTT
	//U   U NN  N MM MM O   O U   U NN  N   T
	//U   U N N N M M M O   O U   U N N N   T
	//U   U N  NN M   M O   O U   U N  NN   T
	// UUU  N   N M   M  OOO   UUU  N   N   T

    //it('Se desmonta el componente.', () => {
    //    component.unmount();
    //});
});

//Eventos (clics).
describe("[MÉTODOS][Componentes][Common][MenuWithBlur] - Valida los eventos dentro del componente.", () => {
	//Se crea el componente.
	//NOTA [1]: Al utilizar la función "mount" se detona la función "componentDidMount".
    //NOTA [2]: Además, con "mount" se puede utilizar el método "attachTo" para utilizar "document".
    let options: Array<any> = [];
    options.push(<label>Inicio</label>);
    options.push(<label>Información de usuario</label>);
    options.push(<label>Storybook vol. 1</label>);
    options.push(<label>Storybook vol. 2</label>);
    options.push(<label>Storybook vol. 3</label>);
    const child = <div style={{ backgroundColor: '#8394DE', height: '100px', width: '100px' }}/>;
	//Se crea el componente.
    //NOTA: Al utilizar la función 'mount' se detona las siguientes funciones: constructor, componentDidMount y render.
    const dummyPage = <div><button type='button' id='btn-anchor'/><MenuWithBlur title='Title' options={options}>{child}</MenuWithBlur></div>;
    //NOTA: Si el componente hace uso de "cocument.getElementById" (o simplemente de "document"), el componente debe "adjuntarse" a este:
    //https://airbnb.io/enzyme/docs/api/mount.html#arguments
    //NOTA: El problema con esto es el siguiente warning:
    /*
    Warning: render(): Rendering components directly into document.body is discouraged, since its children are often manipulated 
    by third-party scripts and browser extensions. This may lead to subtle reconciliation issues. Try rendering into a container 
    element created for your app.
    */
    
    // AAA  N   N TTTTT EEEEE  SSSS      /  DDDD  EEEEE  SSSS PPPP  U   U EEEEE  SSSS
    //A   A NN  N   T   E     S         /   D   D E     S     P   P U   U E     S
    //AAAAA N N N   T   EEE    SSS     /    D   D EEE    SSS  PPPP  U   U EEE    SSS
    //A   A N  NN   T   E         S   /     D   D E         S P     U   U E         S
    //A   A N   N   T   EEEEE SSSS   /      DDDD  EEEEE SSSS  P      UUU  EEEEE SSSS

    let component;
    beforeEach(() => {
        component = enzyme.mount(dummyPage, { attachTo: document.getElementById('root') });
    });
    afterEach(() => {
        //component.unmount();
    });

    // SSSS IIIII M   M U   U L      AAA   CCCC IIIII  OOO  N   N EEEEE  SSSS
    //S       I   MM MM U   U L     A   A C       I   O   O NN  N E     S
    // SSS    I   M M M U   U L     AAAAA C       I   O   O N N N EEE    SSS
    //    S   I   M   M U   U L     A   A C       I   O   O N  NN E         S
    //SSSS  IIIII M   M  UUU  LLLLL A   A  CCCC IIIII  OOO  N   N EEEEE SSSS

    it('Se revisa que exista 1 objeto tipo "button" y un objeto tipo "MenuWithBlur".', () => {
        let outsideButton = component.find('#btn-anchor');
        let menu = component.find('MenuWithBlur');
        //let insideButton = component.find('#btn-menu-options');
        let insideButton = component.find('#btn-menu-options').hostNodes();
        expect(outsideButton).toHaveLength(1);
        expect(menu).toHaveLength(1);
        expect(insideButton).toHaveLength(1);
    });
	it('Debe mostrar el componente, intentarlo nuevamente y ajustar el tamaño y la posición del documento.', () => {
        //Variables.
        let outsideButton = component.find('#btn-anchor');
        let menu = component.find('MenuWithBlur');
        let insideButton = component.find('#btn-menu-options').hostNodes();
        let label = component.find('label').first();
        
        // SSSS H   H  OOO  W   W
        //S     H   H O   O W   W
        // SSS  HHHHH O   O W W W
        //    S H   H O   O WW WW
        //SSSS  H   H  OOO  W   W
        /*
		//Simulación.
        menu.instance().show();
		//Expectativa.
        expect(menu.instance().state.show).toBe(true);
        expect(menu.instance().state.hide).toBe(false);
        */

        // CCCC L     IIIII  CCCC K   K
        //C     L       I   C     K  K
        //C     L       I   C     KKK
        //C     L       I   C     K  K
        // CCCC LLLLL IIIII  CCCC K   K

        //Simulación.
        insideButton.simulate('click', {});
		//Expectativa.
        expect(menu.instance().state.show).toBe(true);
        expect(menu.instance().state.hide).toBe(false);

        //M   M  OOO  U   U  SSSS EEEEE DDDD   OOO  W   W N   N
        //MM MM O   O U   U S     E     D   D O   O W   W NN  N
        //M M M O   O U   U  SSS  EEE   D   D O   O W W W N N N
        //M   M O   O U   U     S E     D   D O   O WW WW N  NN
        //M   M  OOO   UUU  SSSS  EEEEE DDDD   OOO  W   W N   N
        
        //1ro. Se hace click en el mismo componente.
        let event = {
            target: insideButton.instance()
        }
        menu.instance().handleMouseDown(event);
		//Expectativa.
        //expect(menu.instance().state.show).toBe(true);
        //expect(menu.instance().state.hide).toBe(false);
        //2do. Se hace click fuera del componente para cerrarlo.
        event = {
            target: outsideButton.instance()
        }
        menu.instance().handleMouseDown(event);
		//Expectativa.
        //expect(menu.instance().state.show).toBe(false);
        //expect(menu.instance().state.hide).toBe(true);

        //H   H IIIII DDDD  EEEEE
        //H   H   I   D   D E
        //HHHHH   I   D   D EEE
        //H   H   I   D   D E
        //H   H IIIII DDDD  EEEEE
        
        //Al emular un click fuera del componente, la función para ocultarlo se manda llamar.

        // OOO  PPPP   CCCC IIIII  OOO  N   N
        //O   O P   P C       I   O   O NN  N
        //O   O PPPP  C       I   O   O N N N
        //O   O P     C       I   O   O N  NN
        // OOO  P      CCCC IIIII  OOO  N   N

        //Simulación
        label.simulate('click', {});
		//Expectativa.
        //expect(menu.instance().state.show).toBe(false);
        //expect(menu.instance().state.hide).toBe(true);
	});
});