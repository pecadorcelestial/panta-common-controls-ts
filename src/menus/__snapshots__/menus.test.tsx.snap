// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SNAPSHOT][Componentes][Common][MenuWithBlur] - Menú con efecto "blur" sobre el contenido. Debe pintar correctamente el menú con las opciones configuradas. 1`] = `
.c9 {
  box-sizing: border-box;
  display: block;
  height: auto;
  margin: 0px;
  opacity: 0;
  padding: 0px;
  position: absolute;
  width: auto;
  -webkit-animation: gjKtAh 0.6s ease-out forwards;
  animation: gjKtAh 0.6s ease-out forwards;
}

.c6 {
  display: inline-block;
  fill: #FFF;
  height: 50px;
  margin: 0px;
  vertical-align: top;
  width: 50px;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c4 {
  background-color: #1476FB;
  border: 1px solid #1476FB;
  border-radius: 5px;
  box-sizing: border-box;
  color: #FFF;
  cursor: pointer;
  display: inline-block;
  font-size: 14px;
  font-family: "Open Sans",sans-serif;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  height: 30px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 20px;
  padding: 5px 30px;
  text-align: center;
  width: auto;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c4:hover {
  background-color: #0960D3;
  border: 1px solid #0960D3;
  color: #FFF;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover .c5 {
  fill: #FFF;
}

.c4:disabled {
  background-color: #BFBFBF;
  border: 1px solid #BFBFBF;
  color: #FFF;
  pointer-events: none;
}

.c4:disabled .c5 {
  fill: #FFF;
}

.c4:focus {
  outline: none;
}

.c4 .c5 {
  fill: #FFF;
  height: 18px;
  width: 18px;
}

.c3 {
  border-radius: 50%;
  padding: 1px 0px 0px 0px;
  width: 30px;
}

.c0 {
  box-sizing: border-box;
  height: auto;
  margin: 0px;
  padding: 0px;
  width: 100%;
}

.c1 {
  background-color: #FFF;
  box-sizing: border-box;
  display: inline-block;
  float: left;
  height: 50px;
  margin: 0px;
  padding: 10px;
  width: 100%;
}

.c2 {
  cursor: pointer;
  float: left;
  height: 30px;
  margin: 0px;
  padding: 1px 0px 0px 0px;
  width: 30px;
}

.c7 {
  color: #242424;
  float: left;
  font-family: "Open Sans",sans-serif;
  font-size: 20px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  height: 30px;
  margin: 0px 0px 0px 10px;
  overflow: hidden;
  padding: 5px 0px 0px 0px;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: calc(100% - 40px);
}

.c8 {
  background-color: transparent;
  box-sizing: border-box;
  display: block;
  height: auto;
  margin: 50px 0px 0px 0px;
  min-width: 300px;
  opacity: 0;
  padding: 0px;
  position: absolute;
  z-index: 999;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: 0px 0px;
  -ms-transform-origin: 0px 0px;
  transform-origin: 0px 0px;
  -webkit-animation: unset;
  animation: unset;
}

.c10 {
  box-sizing: border-box;
  display: inline-block;
  float: left;
  height: auto;
  margin: 0px;
  padding: 0px;
  width: 100%;
  -webkit-animation: jRxbiA 0.2s ease forwards;
  animation: jRxbiA 0.2s ease forwards;
}

@media screen and (max-width:767px) {
  .c8 {
    min-width: unset;
    width: 80%;
  }
}

@media screen and (min-width:768px) and (max-width:991px) {
  .c8 {
    min-width: unset;
    width: 80%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <button
        className="c3 c4"
        icon="solidTh"
        id="btn-menu-options"
        onClick={[Function]}
        size="small"
        theme="blue"
      >
        <svg
          alt="solidTh"
          className="c5 c6"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M149.333 56v80c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V56c0-13.255 10.745-24 24-24h101.333c13.255 0 24 10.745 24 24zm181.334 240v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.256 0 24.001-10.745 24.001-24zm32-240v80c0 13.255 10.745 24 24 24H488c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24zm-32 80V56c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.256 0 24.001-10.745 24.001-24zm-205.334 56H24c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24zM0 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H24c-13.255 0-24 10.745-24 24zm386.667-56H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zm0 160H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zM181.333 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24z\\"/>",
            }
          }
          viewBox="0 0 512 512"
          x="0px"
          y="0px"
        />
      </button>
    </div>
    <h1
      className="c7"
    >
      Title
    </h1>
  </div>
  <div
    className="c8"
  >
    <div
      className="c9"
      from="left"
      onClick={[Function]}
      style={
        Object {
          "animationDelay": "0s",
          "height": "40px",
          "top": "0px",
          "width": "100%",
        }
      }
      type="fade"
    >
      <label>
        Inicio
      </label>
    </div>
    <div
      className="c9"
      from="left"
      onClick={[Function]}
      style={
        Object {
          "animationDelay": "0.2s",
          "height": "40px",
          "top": "40px",
          "width": "100%",
        }
      }
      type="fade"
    >
      <label>
        Información de usuario
      </label>
    </div>
    <div
      className="c9"
      from="left"
      onClick={[Function]}
      style={
        Object {
          "animationDelay": "0.4s",
          "height": "40px",
          "top": "80px",
          "width": "100%",
        }
      }
      type="fade"
    >
      <label>
        Storybook vol. 1
      </label>
    </div>
    <div
      className="c9"
      from="left"
      onClick={[Function]}
      style={
        Object {
          "animationDelay": "0.6s",
          "height": "40px",
          "top": "120px",
          "width": "100%",
        }
      }
      type="fade"
    >
      <label>
        Storybook vol. 2
      </label>
    </div>
    <div
      className="c9"
      from="left"
      onClick={[Function]}
      style={
        Object {
          "animationDelay": "0.8s",
          "height": "40px",
          "top": "160px",
          "width": "100%",
        }
      }
      type="fade"
    >
      <label>
        Storybook vol. 3
      </label>
    </div>
  </div>
  <div
    className="c10"
  >
    <div
      style={
        Object {
          "backgroundColor": "#8394DE",
          "height": "100px",
          "width": "100px",
        }
      }
    />
  </div>
</div>
`;

exports[`[SNAPSHOT][Componentes][Common][MenuWithBlur] - Menú con efecto "blur" y tema "IENTC" sobre el contenido. Debe pintar correctamente el menú con las opciones configuradas. 1`] = `
.c9 {
  box-sizing: border-box;
  display: block;
  height: auto;
  margin: 0px;
  opacity: 0;
  padding: 0px;
  position: absolute;
  width: auto;
  -webkit-animation: gjKtAh 0.6s ease-out forwards;
  animation: gjKtAh 0.6s ease-out forwards;
}

.c6 {
  display: inline-block;
  fill: #FFF;
  height: 50px;
  margin: 0px;
  vertical-align: top;
  width: 50px;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c11:hover .c5 {
  fill: #FFF;
}

.c11:disabled .c5 {
  fill: #FFF;
}

.c11 .c5 {
  fill: #FFF;
  height: 18px;
  width: 18px;
}

.c4 {
  background-color: #FF0000;
  border: 1px solid #FF0000;
  border-radius: 5px;
  box-sizing: border-box;
  color: #FFF;
  cursor: pointer;
  display: inline-block;
  font-size: 14px;
  font-family: "Open Sans",sans-serif;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  height: 30px;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 20px;
  padding: 5px 30px;
  text-align: center;
  width: auto;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c4:hover {
  background-color: #CC0000;
  border: 1px solid #CC0000;
  color: #FFF;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover .c5 {
  fill: #FFF;
}

.c4:disabled {
  background-color: #BFBFBF;
  border: 1px solid #BFBFBF;
  color: #FFF;
  pointer-events: none;
}

.c4:disabled .c5 {
  fill: #FFF;
}

.c4:focus {
  outline: none;
}

.c4 .c5 {
  fill: #FFF;
  height: 18px;
  width: 18px;
}

.c3 {
  border-radius: 50%;
  padding: 1px 0px 0px 0px;
  width: 30px;
}

.c0 {
  box-sizing: border-box;
  height: auto;
  margin: 0px;
  padding: 0px;
  width: 100%;
}

.c1 {
  background-color: #242424;
  box-sizing: border-box;
  display: inline-block;
  float: left;
  height: 50px;
  margin: 0px;
  padding: 10px;
  width: 100%;
}

.c2 {
  cursor: pointer;
  float: left;
  height: 30px;
  margin: 0px;
  padding: 1px 0px 0px 0px;
  width: 30px;
}

.c7 {
  color: #FFF;
  float: left;
  font-family: "Open Sans",sans-serif;
  font-size: 20px;
  font-stretch: normal;
  font-style: normal;
  font-weight: bold;
  height: 30px;
  margin: 0px 0px 0px 10px;
  overflow: hidden;
  padding: 5px 0px 0px 0px;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: calc(100% - 40px);
}

.c8 {
  background-color: transparent;
  box-sizing: border-box;
  display: block;
  height: auto;
  margin: 50px 0px 0px 0px;
  min-width: 300px;
  opacity: 0;
  padding: 0px;
  position: absolute;
  z-index: 999;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: 0px 0px;
  -ms-transform-origin: 0px 0px;
  transform-origin: 0px 0px;
  -webkit-animation: unset;
  animation: unset;
}

.c10 {
  box-sizing: border-box;
  display: inline-block;
  float: left;
  height: auto;
  margin: 0px;
  padding: 0px;
  width: 100%;
  -webkit-animation: jRxbiA 0.2s ease forwards;
  animation: jRxbiA 0.2s ease forwards;
}

@media screen and (max-width:767px) {
  .c8 {
    min-width: unset;
    width: 80%;
  }
}

@media screen and (min-width:768px) and (max-width:991px) {
  .c8 {
    min-width: unset;
    width: 80%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <button
        className="c3 c4"
        icon="solidTh"
        id="btn-menu-options"
        onClick={[Function]}
        size="small"
        theme="IENTC"
      >
        <svg
          alt="solidTh"
          className="c5 c6"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M149.333 56v80c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V56c0-13.255 10.745-24 24-24h101.333c13.255 0 24 10.745 24 24zm181.334 240v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.256 0 24.001-10.745 24.001-24zm32-240v80c0 13.255 10.745 24 24 24H488c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24zm-32 80V56c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.256 0 24.001-10.745 24.001-24zm-205.334 56H24c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24zM0 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H24c-13.255 0-24 10.745-24 24zm386.667-56H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zm0 160H488c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H386.667c-13.255 0-24 10.745-24 24v80c0 13.255 10.745 24 24 24zM181.333 376v80c0 13.255 10.745 24 24 24h101.333c13.255 0 24-10.745 24-24v-80c0-13.255-10.745-24-24-24H205.333c-13.255 0-24 10.745-24 24z\\"/>",
            }
          }
          viewBox="0 0 512 512"
          x="0px"
          y="0px"
        />
      </button>
    </div>
    <h1
      className="c7"
    >
      Title
    </h1>
  </div>
  <div
    className="c8"
  >
    <div
      className="c9"
      from="left"
      onClick={[Function]}
      style={
        Object {
          "animationDelay": "0s",
          "height": "40px",
          "top": "0px",
          "width": "100%",
        }
      }
      type="fade"
    >
      <label>
        Inicio
      </label>
    </div>
    <div
      className="c9"
      from="left"
      onClick={[Function]}
      style={
        Object {
          "animationDelay": "0.2s",
          "height": "40px",
          "top": "40px",
          "width": "100%",
        }
      }
      type="fade"
    >
      <label>
        Información de usuario
      </label>
    </div>
    <div
      className="c9"
      from="left"
      onClick={[Function]}
      style={
        Object {
          "animationDelay": "0.4s",
          "height": "40px",
          "top": "80px",
          "width": "100%",
        }
      }
      type="fade"
    >
      <label>
        Storybook vol. 1
      </label>
    </div>
    <div
      className="c9"
      from="left"
      onClick={[Function]}
      style={
        Object {
          "animationDelay": "0.6s",
          "height": "40px",
          "top": "120px",
          "width": "100%",
        }
      }
      type="fade"
    >
      <label>
        Storybook vol. 2
      </label>
    </div>
    <div
      className="c9"
      from="left"
      onClick={[Function]}
      style={
        Object {
          "animationDelay": "0.8s",
          "height": "40px",
          "top": "160px",
          "width": "100%",
        }
      }
      type="fade"
    >
      <label>
        Storybook vol. 3
      </label>
    </div>
  </div>
  <div
    className="c10"
  >
    <div
      style={
        Object {
          "backgroundColor": "#8394DE",
          "height": "100px",
          "width": "100px",
        }
      }
    />
  </div>
</div>
`;
